### PR TITLE
New wizard item: Demonic chainsaw

### PR DESCRIPTION
**Pull request by cuackles**
# Document the changes in your pull request
Adds a new item to the wizard spellbook, a demonic chainsaw based on pochita from chainsaw man, this chainsaw gives you lifesteal and does 30 damage each hit with no armor penetration, cost 3 wizard points like mjolmir or however its spelled

2022-05-17 19_11_07-Space station 13_ MacRagge Object Sierra

# Wiki Documentation

add to the wizard items page the chainsaw, noting that it has lifesteal and a price of 3 points

# Changelog
:cl:  
rscadd: Added new wizard item: demon chainsaw
imageadd: added some icons for demon chainsaw
/:cl:

### CODE REVIEW

**Comment by ynot01** [comment]
3 points and no armor pen? yeesh

**Comment by cuackles** [comment]
> 3 points and no armor pen? yeesh

From my testing the health Regen worked pretty well so i didn't want to give it to much, but if it shows to be to weak i can give it a bit of AP or reduce the cost

File: code/game/objects/items/twohanded.dm [hunk 1]
```diff
@@ -677,6 +677,28 @@
...
+	desc = "anime reference here"
+	item_state = "demon_off"
+	icon_state = "demon_off"
+	lefthand_file = 'icons/mob/inhands/weapons/chainsaw_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/chainsaw_righthand.dmi'
+	force = 13
+	force_on = 30
+
+
+/obj/item/twohanded/required/chainsaw/attack_self(mob/user)
```
Comment (tattax) [inline code/game/objects/items/twohanded.dm]:
```suggestion
/obj/item/twohanded/required/chainsaw/demon/attack_self(mob/user)
```

File: code/modules/antagonists/wizard/equipment/spellbook.dm [hunk 2]
```diff
@@ -434,6 +434,11 @@
 	desc = "A hammer that creates an intensely powerful field of gravity where it strikes, pulling everything nearby to the point of impact."
 	item_path = /obj/item/twohanded/singularityhammer
 
+/datum/spellbook_entry/item/demon_chainsaw
+	name = "Demonic Chainsaw"
+	desc = "A demon with the form of a chainsaw,this demon desires to see you fufill your wishes, so they shall give you ethernal life as long as you feed them blood."
```
Comment (adamsong) [inline code/modules/antagonists/wizard/equipment/spellbook.dm]:
```suggestion
	desc = "A demon with the form of a chainsaw, this demon desires to see you fulfill your wishes, so they shall give you eternal life as long as you feed them blood."
```

**Review by adamsong — DISMISSED** [review]

**Comment by ghost** [comment]
It is spelled mjolnir

**Comment by ghost** [comment]
I suggest moving this item code to code/modules/antagonist/wizard/equipment/artifacts

File: code/game/objects/items/twohanded.dm [hunk 3]
```diff
@@ -677,6 +677,28 @@
...
+	icon_state = "demon_off"
+	lefthand_file = 'icons/mob/inhands/weapons/chainsaw_lefthand.dmi'
+	righthand_file = 'icons/mob/inhands/weapons/chainsaw_righthand.dmi'
+	force = 13
+	force_on = 30
+
+
+/obj/item/twohanded/required/chainsaw/demon/attack_self(mob/user)
+	on = !on
+	to_chat(user, "As you pull the starting cord dangling from [src], [on ? "it begins to whirr." : "the chain stops moving."]")
```
Comment (Anvilman6) [inline code/game/objects/items/twohanded.dm:693]: You don't typically pull the starting cord to _stop_ a chainsaw, but this one is also magic, so maybe this is a who cares moment

File: code/modules/antagonists/wizard/equipment/spellbook.dm [hunk 4]
```diff
@@ -434,6 +434,11 @@
 	desc = "A hammer that creates an intensely powerful field of gravity where it strikes, pulling everything nearby to the point of impact."
 	item_path = /obj/item/twohanded/singularityhammer
 
+/datum/spellbook_entry/item/demon_chainsaw
+	name = "Demonic Chainsaw"
+	desc = "A demon with the form of a chainsaw, this demon desires to see you fulfill your wishes, so they shall give you eternal life as long as you feed them blood."
```
Comment (Anvilman6) [inline code/modules/antagonists/wizard/equipment/spellbook.dm]:
```suggestion
	desc = "A demon that has taken the form of a chainsaw! This demon will grant you eternal life, so long as you feed it blood!"
```

**Review by Anvilman6 — CHANGES_REQUESTED** [review]
Small changes

**Review by JamieD1 — APPROVED** [review]

**Pull request merged**